sunpy/net/helioviewer.py

```diff
@@ -54,6 +54,40 @@ def get_data_sources(self, **kwargs):
         params = {"action": "getDataSources"}
         params.update(**kwargs)
         return self._get_json(params)
+    
+    def get_source_id(self):
+        """
+        Returns a dictionary containing the source ID of all the
+        images in the data source
+        """
+        d = dict()
+        from sunpy.net.helioviewer import HelioviewerClient
+        hv = HelioviewerClient()  # doctest: +REMOTE_DATA
+        datasources = hv.get_data_sources() # doctest: +REMOTE_DATA
+        for name, observatory in datasources.items():  # doctest: +REMOTE_DATA
+            if name == "TRACE":  # doctest: +REMOTE_DATA
+                for instrument, params in observatory.items():  # doctest: +REMOTE_DATA
+                    d[(name, instrument)] = params['sourceId']  # doctest: +REMOTE_DATA
+            else:  # doctest: +REMOTE_DATA
+                for inst, detectors in observatory.items():  # doctest: +REMOTE_DATA
+                    for wavelength, params in detectors.items():  # doctest: +REMOTE_DATA
+                        if name in ["Hinode", "STEREO_A", "STEREO_B"] or wavelength in ["C2", "C3"]:  # doctest: +REMOTE_DATA
+                            for wave, adict in params.items():  # doctest: +REMOTE_DATA
+                                d[(name, inst, wavelength, wave)] = adict['sourceId']  # doctest: +REMOTE_DATA
+                        else:  # doctest: +REMOTE_DATA
+                            d[(name, inst, wavelength)] = params['sourceId']  # doctest: +REMOTE_DATA
+
+        return(d)
+
+    def display_source_id(self):
+        """
+        Function to display the source ID of all the images
+        in the data source.
+        """ 
+        from sunpy.net.helioviewer import HelioviewerClient
+        hv = HelioviewerClient()  # doctest: +REMOTE_DATA
+        d_source_id = hv.get_source_id() # doctest: +REMOTE_DATA
+        print("{" + "\n".join("{}: {}".format(k, v) for k, v in d_source_id.items()) + "}")
```

Comment:
What's the advantage to print it with `{}`?

Reply:
I am not sure we need both functions. 

We don't need a function that prints out a dictionary.

Reply:
Yes, this function can be removed.

Reply:
We use that function to display the dictionary.
If anyone wants to see the dictionary they can just use this function

Reply:
You can print a dictionary. We don't need a function to do it.

Reply:
Ok I will do it